homeassistant/components/legrand_rflc/light.py

```diff
@@ -0,0 +1,180 @@
+        await self._async_switch(False)
+
+
+class _LegrandRflcDimmer(_LegrandRflcSwitch):
+    _attr_color_mode = COLOR_MODE_BRIGHTNESS
+    _attr_supported_color_modes = {COLOR_MODE_BRIGHTNESS}
+    _attr_supported_features = SUPPORT_TRANSITION
+
+    @staticmethod
+    def _normalize(value: int, ceiling: int) -> float:
+        """Normalize [0, ceiling] to [0.0, 1.0]."""
+        return value / ceiling
+
+    @staticmethod
+    def _quantize(value: float, ceiling: int) -> int:
+        """Quantize [0.0, 1.0] to [0, ceiling]."""
+        return int(value * ceiling)
+
+    # dimming ceilings for homeassistant and us (LC7001)
+    HA: Final[int] = 255
+    US: Final[int] = 100
+
+    @staticmethod
+    def _to_ha(value: int) -> int:
+        return _LegrandRflcDimmer._quantize(
+            _LegrandRflcDimmer._normalize(value, _LegrandRflcDimmer.US),
+            _LegrandRflcDimmer.HA,
+        )
+
+    @staticmethod
+    def _from_ha(value) -> int:
+        return _LegrandRflcDimmer._quantize(
+            _LegrandRflcDimmer._normalize(value, _LegrandRflcDimmer.HA),
+            _LegrandRflcDimmer.US,
+        )
+
+    def __init__(self, hub, zid: int, properties: Mapping):
+        super().__init__(hub, zid, properties)
+        self._attr_brightness = self._to_ha(properties[hub.POWER_LEVEL])
+
+    async def _zone_properties_changed(self, message: Mapping) -> None:
+        self._zone_properties_changed_switch(message)
+        hub = self._hub
+        if hub.PROPERTY_LIST in message:
+            properties = message[hub.PROPERTY_LIST]
+            if hub.POWER_LEVEL in properties:
+                self._attr_brightness = self._to_ha(properties[hub.POWER_LEVEL])
+        self.async_write_ha_state()
+
+    async def _async_dimmer(self, power: bool, **kwargs) -> None:
+        hub = self._hub
+
+        async def handle(message: Mapping) -> None:
+            hub.StatusError(message).raise_if()
+
+        properties: dict = {"power": power}
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = self._from_ha(kwargs[ATTR_BRIGHTNESS])
+            properties["power_level"] = brightness
+        else:
+            if power:
+                brightness = self._from_ha(self.brightness)
+            else:
+                brightness = 0
+        if ATTR_TRANSITION in kwargs:
+            change = abs(brightness - self._from_ha(self.brightness))
+            properties["ramp_rate"] = min(
+                max(int(change / kwargs[ATTR_TRANSITION]), 1), 100
+            )
+        await hub.handle_send(
+            handle,
+            hub.compose_set_zone_properties(self._zid, **properties),
+        )
+
+    async def async_turn_on(self, **kwargs) -> None:
+        await self._async_dimmer(True, **kwargs)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        await self._async_dimmer(False, **kwargs)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities,
```

Comment:
Please type the whole signature when adding type annotations.

Reply:
Is this what you mean?

--- a/homeassistant/components/legrand_rflc/light.py
+++ b/homeassistant/components/legrand_rflc/light.py
@@ -3,7 +3,7 @@
 from collections.abc import Mapping
 from typing import Final
 
-from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS,
     ATTR_TRANSITION,
@@ -18,7 +18,7 @@ from .const import DOMAIN
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    entry: config_entries.ConfigEntry,
+    entry: ConfigEntry,
     async_add_entities,
 ) -> None:
     """Set up the Legrand RFLC integration light platform.

Reply:
We've already typed the `entry` parameter as `config_entries.ConfigEntry`. That's ok. We haven't typed `async_add_entities`. We should add a type to that parameter.

Reply:
Thanks.
https://github.com/home-assistant/core/pull/53430/commits/ac40a0320407365a1e6494772ec16c9ff884f8ab